tf_big/cc/big_tensor.h

```diff
@@ -48,6 +51,27 @@ typedef Matrix<mpz_class, Dynamic, Dynamic> MatrixXm;
 
 namespace tf_big {
 
+namespace gmp_utils {
+inline void reseed(unsigned long& seed) {
+  int file;
+  /* re-seed random number generator */
+  if ((file = open("/dev/urandom", O_RDONLY)) == -1) {
+    (void)fprintf(stderr, "Error opening /dev/urandom\n");
+  } else {
+    if (read(file, &seed, sizeof seed) == -1) {
+      (void)fprintf(stderr, "Error reading from /dev/urandom\n");
+      (void)close(file);
+    }
+  }
+}
+inline void init_randstate(gmp_randstate_t& state) {
+  gmp_randinit_mt(state);
+  unsigned long seed;
+  reseed(seed);
+  gmp_randseed_ui(state, seed);
+}
+}  // namespace gmp_utils
+
```

Comment:
is this related to capeprivacy/planning#1394? maybe the name/description for the PR should be updated in that case?